### PR TITLE
Anna Dolan - checks for understanding week two

### PR DESCRIPTION
**Pull request by annadolan**

### CODE REVIEW

File: module-2/backend/week_one.md [hunk 1]
```diff
@@ -5,28 +5,99 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+      erb :index, :locals=>{:name => "Mr. Ed"}
+     end
+      ```
 9. What's the purpose of ERB?
+
+    To allow Ruby code within an html document
+    
 10. Why do I need a development AND test database?
+
+      So that your test data does not affect your real database
```
Comment (case-eee) [inline module-2/backend/week_one.md]: Why is this important?

File: module-2/backend/week_one.md [hunk 2]
```diff
@@ -5,28 +5,99 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+    <%= ruby %> #output of code will appear on page
+    <% ruby %> #runs code but does not show output
+    ```
 15. What's an ORM?
+
+      Object Relational Mapping - Maps objects & attributes to database tables
+      
 16. What's the most commonly used ORM?
+
+      I don't know! :(
```
Comment (case-eee) [inline module-2/backend/week_one.md]: ActiveRecord!

File: module-2/backend/week_one.md [hunk 3]
```diff
@@ -5,28 +5,99 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+
+      Updating a database's schema
+    
 19. When you create a migration, does it automatically modify your database?
+    
+      No, you have to run the migration.
+      
 20. How does a model relate to a database?
+
+      It models an object/table in a database and includes methods on that object.
```
Comment (case-eee) [inline module-2/backend/week_one.md]: 👍

**Comment by case-eee** [comment]
@annadolan good work on this. let me know if you have questions 👍

File: module-2/backend/week_two.md [hunk 4]
```diff
@@ -5,37 +5,118 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+
+  A: A migration updates your schema with the changes that you want to make to your database structure.
+    
 3. How does a table relate to a model?
+
+  A: A model represents a table in a database.
+    
 4. What kind of methods are `belongs_to`, and `has_many`? (i.e. class or instance) Give an example.
+
+  A: Class method - e.g. belongs_to :tasks.
```
Comment (s-espinosa) [inline module-2/backend/week_two.md]: Believe this is technically correct (we might need to reword), but per Casey's lesson today the bigger point that we're trying to make is that you call the methods that these relationships offer you on a specific instance. E.g.:

```ruby
blue_dog   = Dog.find(1) # finds a specific *instance* of a dog.
red_house = blue_dog.house # finds the house associated with the blue_dog
```

File: module-2/backend/week_two.md [hunk 5]
```diff
@@ -5,37 +5,118 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+  
 5. What do they allow you to do?
+
+  A: They connect tables in a database through a foreign key.
+  
 6. What's the difference between agile workflow and waterfall method?
+
+  A: Agile workflow breaks the work up into chunks/iterations, tests and refactors each iteration along the way. Waterfall method does each piece of work separately (design, code, test, etc.) one after another.
```
Comment (s-espinosa) [inline module-2/backend/week_two.md]: 👍

File: module-2/backend/week_two.md [hunk 6]
```diff
@@ -5,37 +5,118 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+  
 7. What is the difference between `#new` and `#create`?
+
+  A: "new" initializes an object but doesn't save, "create" initializes and saves.
+  
 8. At a basic level, what does cURL allow you to do?
+
+  A: It allows you to put information into a database through the command line.
```
Comment (s-espinosa) [inline module-2/backend/week_two.md]: It makes HTTP requests from the command line. You're totally right that we can use those requests to put information into a database (like we did in Rush Hour).

File: module-2/backend/week_two.md [hunk 7]
```diff
@@ -5,37 +5,118 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+  
 9. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+
+  A: many-to-many
+  
 10. Define foreign key, primary key, and schema.
+  
+    A: Foreign Key: A foreign key in one table refers to the primary key of another, defines relationships between tables.
+       Primary Key: The key that identifies a table without referring to any of its attributes.
```
Comment (s-espinosa) [inline module-2/backend/week_two.md]: I would say that the key identifies a specific *row* on a table.

File: module-2/backend/week_two.md [hunk 8]
```diff
@@ -5,37 +5,118 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+       Schema:  The architecture used to build a database.
+       
 11. Describe the relationship between a foreign key on one table and a primary key on another table.
+
+   A: It defines the relationship between the tables - when a foreign key is included, it means that the table with the foreign key belongs to the table with the primary key it refers to.
```
Comment (s-espinosa) [inline module-2/backend/week_two.md]: 🎉

File: module-2/backend/week_two.md [hunk 9]
```diff
@@ -5,37 +5,118 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+  
 12. What are the parts of an HTTP response?
+
+  A: Header, response code, body
+
 13. `Rack::Test` allows us to test our controllers in isolation. What are some of the methods it gives us to simulate the request/response cycle?
-14. Describe some techniques to make our Sinatra views more DRY. Give an example of when you would use these techniques.
 
+   A: I don't know what that is! :)
```
Comment (s-espinosa) [inline module-2/backend/week_two.md]: Ha! Good catch. We didn't teach controller testing this time around. You'll get it with Josh in Mod 3.

File: module-2/backend/week_two.md [hunk 10]
```diff
@@ -5,37 +5,118 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+
+14. Describe some techniques to make our Sinatra views more DRY. Give an example of when you would use these techniques.
+   
+   A: Taking all logic out of views and putting it into re-usable methods, using a layout template for your views, using partials. You would use these if you had any parts of your views that repeat - a template is good for a header and footer, for example, and a partial would be useful for a re-used element such as a form.
```
Comment (s-espinosa) [inline module-2/backend/week_two.md]: 👍

File: module-2/backend/week_two.md [hunk 11]
```diff
@@ -5,37 +5,118 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
 
 ### Optional Questions
 
 1. Name your five favorite ActiveRecord methods (i.e. methods your models inherit from ActiveRecord) and describe what they do.
 2. Name your three favorite ActiveRecord rake tasks and describe what they do.
+
+    A:  rake db:create - creates your database!
+        rake db:migrate - migrates changes to your database!
+        rake db:reset db:migrate - removes all the crappy data from your database so you can start over with good data!
```
Comment (s-espinosa) [inline module-2/backend/week_two.md]: ❗️

File: module-2/backend/week_two.md [hunk 12]
```diff
@@ -5,37 +5,118 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+
+    A: Give the teacher table a school_id.
+    
 11. Give an example of when you might want to store information besides ids on a join table.
+
+    A: From our homework! Movies and theaters have a many-to-many relationship through showtimes, which has a "time" attribute as well.
+    
 12. Describe and diagram the relationship between patients and doctors.
+
+    A: many-to-many: patient has_many doctors, doctor has_many patients. This would be represented by a patients_doctors join table, which would contain both foreign ID's.
```
Comment (s-espinosa) [inline module-2/backend/week_two.md]: 👍

**Pull request closed without merging**